pkg/apiserver/registry/generic/key.go

```diff
@@ -0,0 +1,128 @@
+package generic
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/validation/path"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+)
+
+type Key struct {
+	Group     string
+	Resource  string
+	Namespace string
+	Name      string
```

Comment:
@DanCech i removed subresource from this struct. it was unclear why it was needed, and it didn't seem to be used, but i could have missed something

Reply:
Yeah, there are things in the k8s code that imply storing subresources is a thing, but I haven't seen anything actually doing that so removing it seems fine.

Reply:
i believe they use the same key based on my understanding of the usage i've seen for custom resource status and scale storage

Reply:
Is there a difference when it comes to RBAC ? Can a user  have the abilty to perform an action via a subresource ... without having the right the update the underlying resource ? 

As an analogy, I might be allowed to get the logs of a pod (resource="pod/logs") without being allowed to fetch the pod manifest ? Or only allowed to update the status (via the /status resource)  ?

Reply:
> Is there a difference when it comes to RBAC

@chaudyg yeah, there would be, but the status and scale are not stored separately from the rest of the object, so the key is the same. storage doesn't normally handle RBAC, so this isn't an issue.

> allowed to get the logs of a pod (resource="pod/logs")

this is where things are a bit weird, because this is a subresource that probably doesn't hit etcd storage. maybe we need to handle RBAC for subresources before storage?

Reply:
Yeah, subresource handlers are going to need to perform their own rbac checks rather than just relying on storage

Reply:
> this is where things are a bit weird, because this is a subresource that probably doesn't hit etcd storage. maybe we need to handle RBAC for subresources before storage?

The subresource might need to fetch the underlying object. For example, in k8s `pod/logs` has to fetch the underlying pod to find it's location ([here](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/rest/log.go#L92)). Another example is the eviction subresource that will fetch pods and pdb and then [delete the pod accordingly](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/storage/eviction.go#L293-L294).  

AFAIK, in k8s the user making the call to the subresource only needs to RBAC for the subresource, not for the unlying resources.  I agree that the subresource handlers are going to need to perform their own rbac checks rather than just relying on storage ... but for this to work with Unified, the user also needs to have permissions for consuming the resources used by the handler. is that ok ? Do with have such fine grain permissions in grafana ?

Reply:
I may be misinterpreting your comment, but I think that in this case we'll enforce the read permission on the resource at the storage layer, then the handler will need to enforce any additional permission for the specific subresource. That would imply that a user can't hit a subresource of a resource that they cannot read, but I think that's a sensible restriction?